test/firtool/print.fir

```diff
@@ -16,7 +16,7 @@ circuit PrintTest:
     ; CHECK-NEXT:     [[COND:%.+]] = comb.and bin [[PRINTF_COND]], %cond : i1
     ; CHECK-NEXT:     sv.if [[COND]] {
     ; CHECK-NEXT:       [[PRINTF_FD:%.+]] = sv.macro.ref.expr @PRINTF_FD_() : () -> i32
-    ; CHECK-NEXT:       sv.fwrite [[PRINTF_FD]], "test %d\0A"(%var) : i32
+    ; CHECK-NEXT:       sv.fwrite [[PRINTF_FD]], "test %0b %0c %0d %0x\0A"(%var, %var, %var, %var) : i32
```

Comment:
As far as I can tell, tools accept it, but IMO `%0c` is weird since `%c` is by definition a single character. I'm not sure if any lint tools would complain.

Reply:
I can't find evidence in the spec that it's allowed, so let's not do it. Good thinking. Changed to not do this for `%c` in: https://github.com/llvm/circt/pull/8354/commits/c667a3ba6c3d3cfa614f4c5e18e01a9f45992c63